### PR TITLE
Remove group role from state abbreviation

### PR DESCRIPTION
**Pull request by JunTaoLuo**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary

- As part of the staging review, it was decided that the group role is not needed for the state abbreviation in addresses.
- Text size will be addressed separately for all detail page components.
- This is not hidden behind a feature flag
- Appointments (VAOS) team

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/91722

## Testing done

- Tested locally, see screenshot

## Screenshots

After:
image

## What areas of the site does it impact?

Appointments (VAOS)

## Acceptance criteria

For all affected appointment types:
- [ ] Remove `role=group` from the span that wraps the state abbreviation
- [ ] ~~Font size for the content under "Where to attend" section should be 16.96px~~

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/vaos/components/State.jsx [hunk 1]
```diff
@@ -70,9 +71,13 @@ export default function State({ state }) {
   }
 
   return (
-    <span role="group">
+    <span>
```
Comment (jenniemc) [inline src/applications/vaos/components/State.jsx:74]: Maybe the outer span can be eliminated as well.

Comment (JunTaoLuo) [reply src/applications/vaos/components/State.jsx:74]: Good point, let me try that

Comment (JunTaoLuo) [reply src/applications/vaos/components/State.jsx]: Removed.

**Review by jenniemc — APPROVED** [review]
LGTM!

**Pull request merged**